File: app/src/main/java/life/homail/calculator/OperatorBtnEventHandler.java
package life.homail.calculator;
import android.view.View;
public class OperatorBtnEventHandler implements View.OnClickListener{
    private CalculatorMain calculatorMain;

    public OperatorBtnEventHandler(CalculatorMain calculatorMain) {
        this.calculatorMain = calculatorMain;
    }

    @Override
    public void onClick(View view){
        this.findBtnAndPerformOperation(view);
    }
    private void findBtnAndPerformOperation(View view){
        if (view.getId()==R.id.minusBtn){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.minusBtn.getText());
        } else if (view.getId()==R.id.plusBtn){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.plusBtn.getText());
        } else if (view.getId()==R.id.divideBtn){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.divideBtn.getText());
        } else if (view.getId()==R.id.multiplyBtn){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.multiplyBtn.getText());
        }
    }
}

File: app/src/main/java/life/homail/calculator/OperatorBtnEventHandler.kt
package life.homail.calculator
import android.view.View
import android.widget.Button
class OperatorBtnEventHandler(private var calculatorMain: CalculatorMain) : View.OnClickListener{
    private var clickedButton:Button?=null;
    override fun onClick(view: View){
        this.clickedButton=findTheButton(view)
        this.setTextOnScreen(this.clickedButton)
    }
    private fun setTextOnScreen(button: Button?){
        this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(button?.text)
    }
    private fun findTheButton(view: View):Button?{
        when (view.id){
         R.id.divideBtn -> return this.calculatorMain.calculatorViews.divideBtn;
         R.id.plusBtn -> return this.calculatorMain.calculatorViews.plusBtn;
         R.id.minusBtn -> return this.calculatorMain.calculatorViews.minusBtn;
         R.id.multiplyBtn -> return this.calculatorMain.calculatorViews.multiplyBtn;
        }
        return null
    }
}